lib/medtronic/processData.js

```diff
@@ -0,0 +1,378 @@
+      var iob = struct.extractBEShort(record.head,5)/40.0;
+      if (duration > 0) {
+        //TODO: put iob in payload?
+
+        if((record.date[2] & 0xC0) == 0x80) {
+          bolus = cfg.builder.makeDualBolus()
+            .with_duration(duration)
+            .with_extended(amount);
+
+            if(programmed !== amount) {
+              // dual bolus was cancelled
+              var actualDuration = Math.round((amount / (programmed * 1.0)) * duration);
+              bolus = bolus.with_expectedExtended(programmed)
+                        .with_expectedDuration(duration)
+                        .with_duration(actualDuration);
+            }
+
+            i+=1; // advance to next bolus for normal portion
+            //TODO: add second index to payload
+            record = bolusRecords[i];
+            amount = struct.extractBEShort(record.head,3)/40.0;
+            programmed = struct.extractBEShort(record.head,1)/40.0;
+            iob = struct.extractBEShort(record.head,5)/40.0;
+
+            bolus = bolus.with_normal(amount);
+            if(programmed !== amount) {
+              bolus = bolus.with_expectedNormal(programmed);
+            }
+        } else {
+            bolus = cfg.builder.makeSquareBolus()
+              .with_duration(duration)
+              .with_extended(amount);
+
+              if(programmed !== amount) {
+                // square bolus was cancelled
+                var actualDuration = Math.round((amount / (programmed * 1.0)) * duration);
+                bolus = bolus.with_expectedExtended(programmed)
+                          .with_expectedDuration(duration)
+                          .with_duration(actualDuration);
+              }
+          }
+
+        } else {
+          // normal bolus
+          bolus = cfg.builder.makeNormalBolus()
+            .with_normal(amount);
+
+          if(programmed !== amount) {
+            bolus = bolus.with_expectedNormal(programmed);
+          }
+        }
+
+      bolus = bolus.with_deviceTime(sundial.formatDeviceTime(record.jsDate))
+          .set('index', record.index)
+          .set('jsDate', record.jsDate);
+
+      cfg.tzoUtil.fillInUTCInfo(bolus, record.jsDate);
+      bolus = bolus.done();
+      postrecords.push(bolus);
+
+    } else {
+      // TODO: handle smaller bolus records
+    }
+
+  };
+
+  return postrecords;
+};
+
+var processPages = function(data, callback) {
+
+  var records = [];
+  var numRecords = 0;
+  var pages = data.pages;
+
+  if(debug && isBrowser) {
+    savePages(pages);
+  }
+
+  for(var i = 0; i < pages.length; i++) {
+    var page = pages[i];
+    console.log('Page',i,common.bytes2hex(page));
+    var index = 0;
+    while(index < 1024) {
+      var record = {};
+      record.head = struct.extractBytes(page,index,2);
+      if(record.head[0] === 0) {
+        console.log('End of page.');
+        break;
+      }
+      record.type = getType(record.head[0]);
+      if(record.type.value === 0x55) {
```

Comment:
Ah, yes - the constant was already there: `RECORD_TYPES.UNKNOWN_55.value`. I've also added a TODO to support this record type properly and not just throw an error if found. (The  record body length is variable, which makes it a bit tricky)